src/ResourceManager/Insights/Commands.Insights/LogsCmdletBase.cs

```diff
@@ -105,19 +105,17 @@ protected virtual TimeSpan GetDefaultQueryTimeRange()
         /// <summary>
         /// Sets the max number of records to fetch
         /// </summary>
-        protected virtual void SetMaxEventsIfPresent(string currentQueryFilter, string name, int value)
+        protected virtual void SetMaxEventsIfPresent(string currentQueryFilter, int value)
         {
-            if (value > 0 && value <= 100000)
-            {
-                this.MaxEvents = value;
-            }
+            // If value is not acceptable this forces the use of the default value
+            this.MaxRecords = (value > 0 && value <= 100000) ? value : 0;
```

Comment:
I know it is odd. 
There are historic reasons to do it that way. They have to do with the fact that this part of the code is shared by two other cmdlets where MaxEvents is not defined as a parameter, and also MaxEvents was added later.
Anyway this is one thing that I might change later to make it a bit less confusing.